python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_base.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+
+from abc import ABC
+from typing import TypeVar
+
+from azure.ai.inference import load_client as load_client_sync
+from azure.ai.inference.aio import ChatCompletionsClient as ChatCompletionsClientAsync
+from azure.ai.inference.aio import EmbeddingsClient as EmbeddingsClientAsync
+from azure.ai.inference.models import ModelInfo, ModelType
+from azure.core.credentials import AzureKeyCredential
+
+from semantic_kernel.connectors.ai.azure_ai_inference.azure_ai_inference_settings import AzureAIInferenceSettings
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+T = TypeVar("T", bound=ChatCompletionsClientAsync | EmbeddingsClientAsync)
+
+
+@experimental_class
+class AzureAIInferenceBase(KernelBaseModel, ABC):
+    """Base class for Azure AI Inference services."""
+
+    def _create_client(self, settings: AzureAIInferenceSettings, client_type: type[T]) -> tuple[T, ModelInfo]:
+        """Create the Azure AI Inference client.
+
+        Client is created synchronously to check the model type before creating the async client.
+        """
+        credential = AzureKeyCredential(settings.api_key.get_secret_value())
+        sync_client = load_client_sync(endpoint=settings.endpoint, credential=credential)
+
+        model_info = sync_client.get_model_info()
+        if model_info.model_type not in self._get_desired_mode_type(client_type):
```

Comment:
if this function is not used anywhere else, you might as well make it a check function that returns a bool...